dist/msrcrypto.js

```diff
@@ -8933,10 +8933,12 @@ var workerManager = (function () {
             }
 
             // Send the results to the operation object and it will fire
-            //   it's onCompleted event.
-            if (op && e.data.type !== "process") {
+            // it's onCompleted event.
+            if (op) {
                 jobCompleted(worker);
-                op.dispatchEvent(e);
+                if(e.data && e.data.type !== "process"){

```

Comment:
Hey @JamieRez, I haven't forgotten about this PR. I just want to ensure this is the logic we want. The alternative is:

```javascript
if(!e.data || e.data.type !== "process")
```

It seems in other places, [the pattern above is used](https://github.com/kevlened/msrCrypto/blob/27815729ceafa1f4e6029fdf0999bd8da83c8752/dist/msrcrypto.js#L8598). I want a better understanding of what this logic is trying to do before merging.

Reply:
Your original logic is correct.

The result of the statement is `op.dispatchEvent` is called, [`function opDispatchEvent` calls `processResults(e.data)`](https://github.com/kevlened/msrCrypto/blob/27815729ceafa1f4e6029fdf0999bd8da83c8752/dist/msrcrypto.js#L8634-L8648) and [`processResults` requires `e.data` to be a non-null Object](https://github.com/kevlened/msrCrypto/blob/27815729ceafa1f4e6029fdf0999bd8da83c8752/dist/msrcrypto.js#L8667-L8671).